### PR TITLE
Implement rolling deployment

### LINKED ISSUE #3950: Use 'buildpacks' instead of deprecated 'buildpack' in manifest files
Use 'buildpacks' instead of deprecated 'buildpack' in manifest files. https://docs.cloudfoundry.org/devguide/deploy-apps/manifest-attributes.html#buildpack-deprecated

I think `autopilot` needs `buildpack` (see https://github.com/concourse/cf-resource/pull/72#issuecomment-476465049) so this is blocked by exploring/implementing autopilot alternatives (#3597)

### LINKED ISSUE #4668: Implement rolling deployments
**What we're after:**
Implement rolling deployments using cf cli version 7 so we can implement CMS/Proxy settings (see https://github.com/fecgov/fec-accounts/issues/289). Change in all repos for consistency/developer experience.

Example: `cf push proxy --strategy rolling -f manifest_dev.yml `

**Completion criteria**
- [x] Read the docs https://docs.cloudfoundry.org/devguide/deploy-apps/rolling-deploy.html
- [x] Move application manifest attributes from top-level to underneath the app (required for rolling)
- [X] Consider deploying multiple applications at once https://docs.cloudfoundry.org/devguide/deploy-apps/manifest.html#multi-apps (not possible with rolling deployments)
- [x] Switch out zero-downtime-deploy with  --strategy rolling
- [x] Handle partially failed deployments with `cf cancel-deployment` commands. 
- [x] Consider printing output from https://docs.cloudfoundry.org/devguide/deploy-apps/rolling-deploy.html#status 
- [x] Remove any references to `autopilot`
- [x] Update README

### PR DESCRIPTION
**Pull request by lbeaufort**
## Summary (required)

Resolves #4668
Resolves #3950 

- Reformat manifests with newer format
- Implement rolling deployments, remove references to `autopilot` and `zero-downtime-push`

## Reviewers
- At least two reviewers please

## How to test the changes locally

**Through CircleCI**
- Make your own copy of this branch (please don't commit to this branch)
- Modify deploy task to deploy to a space of your choice
- Push to Github, make sure build works

**Locally:**
- Install cf cli version [following this guide](https://github.com/cloudfoundry/cli/wiki/Version-Switching-Guide) or:
```
# Install cf-cli@7. Complete commands
brew install cf-cli@7

# If you run across an error like this: Error: No such keg: /usr/local/Cellar/cf-cli@6, make sure you install cf-cli version 6 with brew
brew install cf-cli@6
 
# Unlink cf-cli version 6 and link cf-cli version 7
brew unlink cf-cli@6 && brew unlink cf-cli@7 && brew link cf-cli@7 && cf version
```
- Do a manual deploy with `invoke deploy --space dev`

##  Testing failed builds
- Test with bad requirements.txt, like `Flask=2222`, make sure build fails
- Test with crashing app - add 5 instances,  at the top of `run.sh`, add:
```
rand_mod=$(expr $RANDOM % 5)
if [ $rand_mod -eq 0 ]; then
    echo "Starting divisible by five app"
else
    echo "Crashing non-divisible by five app";
    exit 1
fi
```
make sure build fails and is cancelled

## Impacted areas of the application
List general components of the application that this PR will affect:

-  Deploy task only

### CODE REVIEW

File: manifests/manifest_celery_worker_stage.yml [hunk 1]
```diff
@@ -20,3 +7,16 @@ applications:
     no-route: true
     health-check-type: process
     command: celery worker --app webservices.tasks --loglevel ${LOGLEVEL:=INFO} --concurrency 2
+    path: ../
+    stack: cflinuxfs3
+    buildpack:
```
Comment (lbeaufort) [inline manifests/manifest_celery_worker_stage.yml:12]: Missed one `buildpack`

File: manifests/manifest_celery_worker_prod.yml [hunk 2]
```diff
@@ -21,3 +7,17 @@ applications:
     no-route: true
     health-check-type: process
     command: celery worker --app webservices.tasks --loglevel ${LOGLEVEL:=INFO} --concurrency 3
+    path: ../
+    stack: cflinuxfs3
+    buildpack:
```
Comment (lbeaufort) [inline manifests/manifest_celery_worker_prod.yml:12]: Another `buildpack`

File: manifests/manifest_celery_worker_dev.yml [hunk 3]
```diff
@@ -18,3 +7,14 @@ applications:
     no-route: true
     health-check-type: process
     command: celery worker --app webservices.tasks --loglevel ${LOGLEVEL:=INFO} --concurrency 2
+    path: ../
+    stack: cflinuxfs3
+    buildpack:
```
Comment (lbeaufort) [inline manifests/manifest_celery_worker_dev.yml:12]: +1

**Comment by lbeaufort** [comment]
@jason-upchurch @fec-jli @pkfec I added success messages and the app status to all three PRs:

- https://github.com/fecgov/openFEC/pull/4671/commits/067bf4d6513136e4c9267bcec15c21a372aa6059
- https://github.com/fecgov/fec-cms/pull/4148/commits/6b5545496472e018826ae4ff4540bfe4d5d57043
- https://github.com/fecgov/fec-eregs/pull/537/commits/c2f361b7e7204ee92e9a2f179f8060b2f93b2903

For the API, we're pushing 3 apps and I always found that message confusing, so I changed the message to output _which_ application was successfully pushed.

There's probably some room for improvement on spacing, but since this is blocking a time-sensitive issue I think we can address those in the next iteration.

**Review by jason-upchurch — APPROVED** [review]
Thanks for including the explicit success output @lbeaufort ! I think it will help devs QA the build/deploy in a less ambiguous way. We appreciate all of your hard work and coordination with cloud.gov to implement this! Nice work 👍

**Review by fec-jli — APPROVED** [review]
Pair with Jason and Priya, test on dev space. work as expected.
Great job.

**Pull request merged**